### PR TITLE
LPS-138078 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-138078

cc @david-gutierrez-mesa 
cc @Tim-Cao 
cc @liulukiwi

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 1]
```diff
@@ -1378,6 +1378,486 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-126009. The user is able to filter Web Content in Collection Display using Tag and Title field as parameter."
+	@priority = "4"
+	test CanFilterWebContentInCollectionDisplayUsingTagAndTitleField {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1383]: FilterWebContentByTagAndTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 2]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+		task ("View the Collection Display is empty") {
+			AssertVisible(
+				key_infoMessage = "The collection is empty. To display your items, add them to the collection or choose a different collection.",
+				locator1 = "Message#INFO_SPECIFIC");
+		}
+	}
+
+	@description = "This is a test for LPS-126009. The user is able to filter Web Content in Collection Display using Tag field as parameter."
+	@priority = "4"
+	test CanFilterWebContentInCollectionDisplayUsingTagField {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1575]: FilterWebContentByTag

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 3]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+					locator1 = "PageEditor#COLLECTION_ITEM_TITLE");
+
+				var n = ${n} + 1;
+			}
+		}
+	}
+
+	@description = "This is a test for LPS-126009. The user is able to filter Web Content in Collection Display using Title field as parameter."
+	@priority = "4"
+	test CanFilterWebContentInCollectionDisplayUsingTitleField {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1740]: FilterWebContentByTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 4]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+			GogoShell.viewOutput(outputContent = "true");
+		}
+
+		task ("Add three tags") {
+			for (var tagName : list "tag1,tag2,tag3") {
+				JSONAssettag.addTag(
+					groupName = "Test Site Name",
+					tagName = "${tagName}");
+			}
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1400]: Redundant steps

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 5]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1592]: Redundant steps

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 6]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+
+		task ("Add three Web Contents") {
+			for (var n : list "1,2,3") {
+				JSONWebcontent.addWebContent(
+					content = "Web Content Content ${n}",
+					groupName = "Test Site Name",
+					title = "Web Content Title ${n}");
+			}
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1758]: Redundant steps

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 7]
```diff
@@ -596,6 +596,50 @@ definition {
...
+				key_text = "Clear",
+				locator1 = "Button#ANY");
+		}
+
+		if (isSet(tagNames)) {
+			Click(
+				key_text = "Select",
+				locator1 = "Button#ANY");
+
+			for (var tagList : list "${tagNames}") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:615]: for (var tagName : list "${tagNameList}")

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 8]
```diff
@@ -596,6 +596,50 @@ definition {
...
+			fieldLabel = "Collection",
+			menuItem = "Filter Collection");
+
+		if (isSet(clear)) {
+			Click(
+				key_text = "Clear",
+				locator1 = "Button#ANY");
+		}
+
+		if (isSet(tagNames)) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:610]: tagNameList

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 9]
```diff
@@ -596,6 +596,50 @@ definition {
...
+				locator1 = "Button#ANY");
+
+			for (var tagList : list "${tagNames}") {
+				Check.checkNotVisible(
+					checkboxName = "${tagList}",
+					locator1 = "Checkbox#ANY_CHECKBOX");
+			}
+		}
+
+		if (isSet(titleField)) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:622]: title

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Message.path [hunk 10]
```diff
@@ -250,6 +250,11 @@
 	<td>//div[@role='alert' and contains(.,'Please select a valid date range.')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>FILTER_COLLECTION_INFO</td>
+	<td>//div[contains(@class,'tbar-section')]/span[contains(.,'${key_infoMessage}')]</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Message.path:257]: Move to Modal.path and Rename to SEARCH_INFO
`//div[contains(@class,'modal-dialog')]//div[contains(@class,'container-fluid')]//span[contains(@class,'text-truncate')]`

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 11]
```diff
@@ -596,6 +596,50 @@ definition {
...
+			Type(
+				key_text = "Title",
+				locator1 = "TextInput#ANY",
+				value1 = "${titleField}");
+
+			Click(locator1 = "Modal#HEADER");
+		}
+
+		if (isSet(infoMessage)) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:631]: if (isSet(filterResults))

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 12]
```diff
@@ -596,6 +596,50 @@ definition {
...
+			AssertTextEquals(
+				key_infoMessage = "${infoMessage}",
+				locator1 = "Message#FILTER_COLLECTION_INFO",
+				value1 = "${infoMessage}");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:635]: ${filterResults}

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 13]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+				infoMessage = '''There are 0 results for "Web Content Title 3", tag3.''',
+				tagNames = "tag3");
+		}
+
+		task ("View the Collection Display is empty") {
+			AssertVisible(
+				key_infoMessage = "The collection is empty. To display your items, add them to the collection or choose a different collection.",
+				locator1 = "Message#INFO_SPECIFIC");
+		}
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1571]: Add steps to disable feature

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 14]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+				AssertVisible(
+					key_colPosition = "1",
+					key_entryTitle = "Web Content Title ${titleNumber}",
+					key_fragmentName = "Collection Display",
+					key_rowPosition = "${n}",
+					locator1 = "PageEditor#COLLECTION_ITEM_TITLE");
+
+				var n = ${n} + 1;
+			}
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1735]: Add steps to disable feature

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase [hunk 15]
```diff
@@ -1378,6 +1378,486 @@ definition {
...
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1859]: Add steps to disable feature

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update changes according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase (hunk 6 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1758]: Hi @Tim-Cao, could you explain why this step is redundant? If I don't create the web contents, how am I going to use the filter?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase (hunk 5 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1592]: Hi @Tim-Cao, it's the same question that I did in the other comment, I need the tag to use the filter, why is it redudant?

Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1592]: For tags, when create asset, the tags could be auto generated. No need to create tag first

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase (hunk 4 above)
Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1400]: Only the tag3 is needed.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase (hunk 5 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/collectiondisplay/CollectionDisplay.testcase:1592]: Thanks for clarifying

**Comment by ozysouza** [comment]
> Hi @ozysouza, Could you update changes according to comments? Thanks

Hi @Tim-Cao, PR updated following your comments. Thanks for the help!

**Comment by david-gutierrez-mesa** [comment]
@Tim-Cao @ozysouza any update on this?

**Comment by ozysouza** [comment]
> @Tim-Cao @ozysouza any update on this?

Hi @david-gutierrez-mesa, I sent the changes that were requested, I'm just waiting the review to see if there is some other change that will be needed.

**Comment by Tim-Cao** [comment]
Just start reviewing

**Comment by Tim-Cao** [comment]
Hi @ozysouza, there is a SF issue. Btw, see changes https://github.com/Tim-Cao/liferay-portal/compare/9198449ad33e20135d5c99a4d3aa26f239a28488...69a76022b1d422b044264e913198f6524dac9a0b

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**